src/Metadata/RootMetadata.php

```diff
@@ -59,7 +60,7 @@ protected static function getSignedCollectionOptions(): array
      */
     public function getRoles()
     {
-        return $this->getSigned()['roles'];
+        return deep_copy($this->getSigned()['roles']);
```

Comment:
In any of the getters where we are already calling `getSigned()` we don't need to call `deep_copy()` again because `getSigned()` already calls this and gives a us a clone